fix test_client.py tests

AtomsForPeace opened the pull request:
Got one set of tests working with a simple import change. Still haven't gotten the integration tests working.

AtomsForPeace commented:
Doesn't work in python 3.5 as yield isn't allowed in async functions.
Possible workaround is to use: https://github.com/python-trio/async_generator

AtomsForPeace commented:
I recommend removing testing on python3.5 and adding python3.7 instead. The tests will then work without any further changes

cecton commented on .travis.yml:
```diff
@@ -6,14 +6,15 @@ services:
   - docker
 
 python:
-  - "3.5"
```
Is Python 3.5 not desired anymore? If it is a mistake, you can keep 3.5, 3.6 and 3.7

cecton replied:
Just read your comment:

>I recommend removing testing on python3.5 and adding python3.7 instead. The tests will then work without any further changes

Well I believe it would be best if we can make it work so it includes 3.5. Is it possible reasonably or is it too complex?

cecton commented on README.rst:
```diff
@@ -88,11 +88,22 @@ Installation
 Requirements
 ^^^^^^^^^^^^
 
  *  Python >= 3.5
```
(The version requirement should match the tests of the CI)

AtomsForPeace replied:
Agreed

cecton commented on README.rst:
```diff
@@ -88,11 +88,22 @@ Installation
+
+   ::
+   
+      docker run -d --name travis-virtuoso -p 8890:8890 -e SPARQL_UPDATE=true tenforce/virtuoso:1.2.0-virtuoso7.2.2
```
Actually you now need both containers to be started I think. Right?

AtomsForPeace replied:
Thanks, didn't get back to add that yet

cecton commented on conftest.py:
```diff
@@ -0,0 +1,72 @@
+
+@pytest.fixture
+def test_client():
+    return AioSPARQLTestCase()
```
This fixture doesn't seem to be used. (Also, a TestCase is not a Client.)

AtomsForPeace replied:
Removed

cecton commented on conftest.py:
```diff
@@ -0,0 +1,72 @@
+
+
+@pytest.fixture
+async def jena_client(loop, jena_endpoint):
```
You don't seem to need `loop`. Do you need to specify it in argument?

cecton commented on conftest.py:
```diff
@@ -0,0 +1,72 @@
+
+
+@pytest.fixture
+async def virtuoso_client(loop, virtuoso_endpoint):
```
You don't seem to need `loop`. Do you need to specify it in argument?

cecton commented on tests/unit/test_client.py:
```diff
@@ -2,12 +2,14 @@
+)
 from aiosparql.syntax import IRI, RDF, Triples
-from aiosparql.test_utils import AioSPARQLTestCase, unittest_run_loop
+from aiosparql.test_utils import AioSPARQLTestCase
```
The import is not used anymore. Can you add something to the CI to check for not used imports? I don't know if it's complicated. Otherwise just remove it.

cecton commented on aiosparql/test_utils.py:
```diff
@@ -78,7 +74,7 @@ def post(self, *args, **kwargs):
 
 
-class AioSPARQLTestCase(AioHTTPTestCase):
+class AioSPARQLTestCase(TestSPARQLClient):
```
I don't think this was intended to be used that way.  TestSPARQLClient is a client class while AioSPARQLTestCase is a TestCase class for unittest. Plus I don't think this class is used anywhere, you can maybe remove it entirely.

AtomsForPeace replied:
Yeah, I didn't do much in this area. Not sure what's best.

cecton commented on aiosparql/client.py:
```diff
@@ -182,6 +182,8 @@ def _pretty_print_query(self, query: str) -> str:
                     explanation=explanation)
+            if resp.content_type == 'text/html':
+                return resp
             return await resp.json()
```
I don't think the return type matches. resp is probably an object response and the return type expected is a `dict`.

AtomsForPeace replied:
Good point! I say your commit in add jena support and that's much better

cecton commented:
I believe we can close it in favor of #38 and #39 

@AtomsForPeace thanks for the contribution!

asvetlov commented:
@cecton I've filed #40

AtomsForPeace commented:
So I removed the unrelated stuff from this pull request. For example the tests fail at the moment because there's no support for jena. We can add that in the jena branch. I also removed testing on py35 because I saw that @cecton added that in the jena branch. I suggest we get the jena branch done and merge it into this branch. Or am I missing something? Thanks for the help

AtomsForPeace commented:
Or is everything already covered in #38? I see the conftest isn't there and a few other things

asvetlov commented:
jena branch is merged into master.
Please feel free to update the PR

cecton commented:
> 
> 
> Or is everything already covered in #38? I see the conftest isn't there and a few other things

@AtomsForPeace  Yes! As I mentioned [here](https://github.com/aio-libs/aiosparql/pull/37#issuecomment-539997690), I have merged your changes on 8c9014acc27974c5b1a1b64e890e682d433e1342 mostly unchanged but we lost the author of the commit sorry.

cecton commented:
(The build on master is red but I think it's a false negative)

AtomsForPeace commented:
Ok, thanks for merging. I'll close it now
